lib/project_types/script/layers/infrastructure/languages/assemblyscript_task_runner.rb

```diff
@@ -49,15 +55,19 @@ def metadata
 
           private
 
-          def check_node_version!
-            output, status = @ctx.capture2e("node", "--version")
-            raise Errors::DependencyInstallError, output unless status.success?
+          def check_tool_version!(tool, min_required_verison)
+            output, status = @ctx.capture2e(tool, "--version")
+            unless status.success?
+              raise Errors::MissingDependencyError,
+                "#{tool} version must be >= v#{min_required_verison}. "\
+                "No version of #{tool} installed."
+            end
 
             require "semantic/semantic"
-            version = ::Semantic::Version.new(output[1..-1])
-            unless version >= ::Semantic::Version.new(AssemblyScriptProjectCreator::MIN_NODE_VERSION)
-              raise Errors::DependencyInstallError,
-                "Node version must be >= v#{AssemblyScriptProjectCreator::MIN_NODE_VERSION}. "\
+            version = ::Semantic::Version.new(output.gsub(/^v/, ""))
+            unless version >= ::Semantic::Version.new(min_required_verison)
+              raise Errors::MissingDependencyError,
+                "#{tool} version must be >= v#{min_required_verison}. "\
```

Comment:
I don't think you need the string here?  All strings are defined in `messages.rb`.  I see you already did that.  :)